Source/spells.cpp

```diff
@@ -296,46 +293,37 @@ void DoResurrect(int pnum, int rid)
 
 void DoHealOther(int pnum, int rid)
 {
-	int i, j, hp;
-
 	if (pnum == myplr) {
 		NewCursor(CURSOR_HAND);
 	}
 
 	auto &player = plr[pnum];
 	auto &target = plr[rid];
 
-	if ((char)rid != -1 && (target._pHitPoints >> 6) > 0) {
-		hp = (GenerateRnd(10) + 1) << 6;
-
-		for (i = 0; i < player._pLevel; i++) {
-			hp += (GenerateRnd(4) + 1) << 6;
-		}
-
-		for (j = 0; j < GetSpellLevel(pnum, SPL_HEALOTHER); ++j) {
-			hp += (GenerateRnd(6) + 1) << 6;
-		}
-
-		if (player._pClass == HeroClass::Warrior || player._pClass == HeroClass::Barbarian) {
-			hp *= 2;
-		} else if (player._pClass == HeroClass::Rogue || player._pClass == HeroClass::Bard) {
-			hp += hp / 2;
-		} else if (player._pClass == HeroClass::Monk) {
-			hp *= 3;
-		}
-
-		target._pHitPoints += hp;
+	if (rid == -1 || (target._pHitPoints >> 6) <= 0) {

```

Comment:
This function is called from `On_HEALOTHER` which passes in a `uint16_t` as the `rid` parameter. This means `rid` should never be -1, but AFAIK it could be after casting to `char`. Either this change introduces potential errors or it's entirely unnecessary.

Reply:
The signature casts it to an `int`, isn't that enough?

Reply:
`DoHealOther` should take `int pnum, uint rid`. Converting from `uint16_t` to `int` is implementation defined. The usual implementation for x86 is simply sign-extending the MSB and reinterpreting as the target type, so this will preserve -1 (really `std::numeric_limits<uint16_t>::max()` in this context) as -1.

Assuming you change the function signature you then avoid IB. The cast to char is unnecessary in both cases. Regarding the comparison with -1, conversion of a signed value to an unsigned value is well defined but you may get warnings from your compiler if it's implicit. If so, wrapping -1 in a cast to uint16_t would supress it but the better solution would be to define a constant along the lines of `constexpr uint16_t NO_TARGET = -1`

Reply:
@ephphatha FYI, I did my testing using MSVC. I thought you might be curious.